File: scripts_divers/D6_to_D7/fix-images-expat.py
#!/usr/bin/env python3

# In the expat website, when we try to migarte it, the images not found
# and the only solution we found is to use this script to patch its database
# and so repair the links

import pymysql as mysql
import re

db = mysql.connect(host="localhost", user="root", passwd="tata", db="expat", charset='utf8')

with db:
    cur = db.cursor()


    ## Update user's pictures
    sql_select = """
    SELECT uid, picture FROM users
    """

    sql_update = """
    UPDATE users SET picture = %s WHERE uid = %s
    """

    cur.execute(sql_select)
    rows = cur.fetchall()
    for uid, picture in rows:
        picture = re.sub(r'sites/assos.centrale-marseille.fr.expat/files/(.*)', r'\1', picture)
        cur.execute(sql_update, (picture, uid))


    ## Update other files
    sql_select = """
    SELECT fid, filepath FROM files
    """

    sql_update = """
    UPDATE files SET filepath = %s WHERE fid = %s
    """

    cur.execute(sql_select)
    rows = cur.fetchall()
    for fid, file_path in rows:
        file_path = re.sub(r'sites/assos.centrale-marseille.fr.expat/files/(.*)', r'\1', file_path)
        cur.execute(sql_update, (file_path, fid))

    db.commit()

File: other-scripts/D6-to-D7/fix-images-expat.pl
# In the expat website, when we try to migarte it, the images not found
# and the only solution we found is to use this script to patch its database
# and so repair the links

use strict;
use warnings;
use DBI;

my $db = 'expat';
my $server = 'localhost';
my $user = 'root';
my $password = 'tata';
my $port = '';

#### Create DB connection
my $dbh = DBI->connect("DBI:mysql:database=$db;host=$server;port=$port", $user, $password, { RaiseError => 1, }) or die "Cannont connect to database $db";

# Update user's pictures

my $sql_select =<<"SQL";
SELECT uid, picture FROM expat_users
SQL

my $sql_update =<<"SQL";
UPDATE expat_users SET picture = ? WHERE uid = ?
SQL

my $req_select = $dbh->prepare($sql_select);
my $req_update = $dbh->prepare($sql_update);
$req_select->execute;

while (my ($uid, $picture) = $req_select->fetchrow_array)
{
    $picture =~ s#sites/assos.centrale-marseille.fr.expat/(.*)#sites/default/$1#;
    $req_update->execute($picture, $uid);
}

$req_update->finish;
$req_select->finish;

# Update other file
$sql_select = "SELECT fid, filepath FROM expat_files";
$sql_update = "UPDATE expat_files SET filepath = ? WHERE fid = ?";

$req_select = $dbh->prepare($sql_select);
$req_update = $dbh->prepare($sql_update);
$req_select->execute;

while (my ($fid, $file_path) = $req_select->fetchrow_array)
{
    $file_path =~ s#sites/assos.centrale-marseille.fr.expat/(.*)#sites/default/$1#;
    $req_update->execute($file_path, $fid);
}

$req_update->finish;
$req_select->finish;

$dbh->disconnect;
